framework/wazuh/manager.py

```diff
@@ -384,3 +384,53 @@ def get_basic_info():
     result.total_affected_items = len(result.affected_items)
 
     return result
+
+
+@expose_resources(actions=[f"{'cluster' if cluster_enabled else 'manager'}:upload_config"],
+                  resources=[f'node:id:{node_id}' if cluster_enabled else '*:*:*'])
+def update_ossec_conf(new_conf=None):
+    """
+    Replaces wazuh configuration (ossec.conf) with the provided configuration.
+
+    Parameters
+    ----------
+    new_conf: str
+        The new configuration to be applied.
+    """
+    result = AffectedItemsWazuhResult(all_msg=f"Configuration was successfully updated"
+                                              f"{' in specified node' if node_id != 'manager' else ''}",
+                                      some_msg='Could not update configuration in some nodes',
+                                      none_msg=f"Could not update configuration"
+                                               f"{' in specified node' if node_id != 'manager' else ''}"
+                                      )
+    backup_conf = None
+    try:
+        # Check a configuration has been provided
+        if new_conf is None or not new_conf or len(new_conf) == 0:
+            raise WazuhError(1125)
+
+        # Check if the configuration is valid
+        new_conf = prettify_xml(new_conf)
+
+        # Create a backup of the current configuration before attempting to replace it
+        backup_conf = load_wazuh_xml(common.ossec_conf)
+        write_ossec_conf(new_conf)
+        is_valid = validate_ossec_conf()
```

Comment:
Can't we validate before writing?

Reply:
No, unfortunately due to the way the Core validation process works we cannot send the configuration itself to the socket. We need to place the new configuration inside the ossec.conf and then request its validation using the socket instead.

Reply:
In the future we should improve this to be able to send the configuration to be validated directly to the socket itself, avoiding unnecesary changes to the ossec.conf and potential issues derived from that.

Reply:
Apply file backup system

Reply:
Added file backup functionality